consensus/types.go

```diff
@@ -135,12 +132,14 @@ func (b Block) CheckTarget(target Target) bool {
 
 // ExpectedTransactionMerkleRoot() returns the expected transaction
 // merkle root of the block.
-func (b Block) TransactionMerkleRoot() hash.Hash {
-	var transactionHashes []hash.Hash
-	for _, transaction := range b.Transactions {
-		transactionHashes = append(transactionHashes, hash.HashBytes(encoding.Marshal(transaction)))
+func (b Block) MerkleRoot() hash.Hash {
+	var hashes []hash.Hash
+	hashes = append(hashes, hash.HashObject(b.Timestamp))
+	hashes = append(hashes, hash.HashObject(b.MinerAddress))
```

Comment:
Not worried about performance right now, just something that's clean and easy to read.

I'd imagine that consecutive appends get optimized by the compiler anyway.